cmd/kubeadm/app/phases/selfhosting/podspec_mutation.go

```diff
@@ -18,38 +18,42 @@ package selfhosting
 
 import (
 	"k8s.io/api/core/v1"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 // mutatePodSpec makes a Static Pod-hosted PodSpec suitable for self-hosting
-func mutatePodSpec(name string, podSpec *v1.PodSpec) {
-	mutators := map[string][]func(*v1.PodSpec){
+func mutatePodSpec(cfg *kubeadmapi.MasterConfiguration, name string, podSpec *v1.PodSpec) {
```

Comment:
Totally agree here. This felt very ugly as I was writing it, but was the only way to get that information to where it needed to go.